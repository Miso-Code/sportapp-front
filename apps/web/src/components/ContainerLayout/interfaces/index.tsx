import { ReactNode } from 'react'

export interface Props {
	children: ReactNode
	className?: string
	secondarySection?: ReactNode
	withSecondarySection?: boolean
}
