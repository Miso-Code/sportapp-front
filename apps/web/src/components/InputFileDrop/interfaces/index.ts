import { Accept } from 'react-dropzone'

export interface Props {
	onChange: (file: File) => void
	sizeFile?: number
	accept?: Accept | undefined
	value?: File | null
	multiple?: boolean
	className?: string
	label?: string
	placeHolder?: string
}
