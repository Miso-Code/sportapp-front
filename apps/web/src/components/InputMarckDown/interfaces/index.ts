export interface Props {
	readonly value: string
	readonly onChange: (value: string | undefined) => void
	readonly label: string
	readonly className?: string
}
