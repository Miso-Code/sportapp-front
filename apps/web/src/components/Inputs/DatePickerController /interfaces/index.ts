import { Control, FieldValues } from 'react-hook-form'

export interface Props {
	control: Control<FieldValues, unknown> | undefined
	name: string
	label: string
	fullWidth?: boolean
}
