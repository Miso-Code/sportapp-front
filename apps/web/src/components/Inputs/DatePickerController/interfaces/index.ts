import { Control, FieldValues, Path } from 'react-hook-form'

export interface Props<T extends FieldValues = FieldValues> {
	control: Control<T> | undefined
	name: Path<T>
	label: string
	fullWidth?: boolean
}
