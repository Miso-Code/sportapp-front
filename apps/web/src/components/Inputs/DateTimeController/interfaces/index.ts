import { Control, FieldValues, Path } from 'react-hook-form'
import { DateTimePickerProps } from '@mui/x-date-pickers/DateTimePicker'

export interface Props<T extends FieldValues = FieldValues>
	extends DateTimePickerProps<Date, boolean> {
	control: Control<T> | undefined
	readonly name: Path<T>
	readonly label: string
	readonly fullWidth?: boolean
}
