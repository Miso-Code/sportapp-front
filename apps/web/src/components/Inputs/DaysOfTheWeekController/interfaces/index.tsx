import { Control, FieldValues, Path } from 'react-hook-form'

export interface Props<T extends FieldValues = FieldValues> {
	control: Control<T> | undefined
	disabled?: boolean
	readonly name: Path<T>
	readonly label: string
}
