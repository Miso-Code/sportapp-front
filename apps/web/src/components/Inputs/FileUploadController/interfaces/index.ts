import { Control, FieldValues, Path } from 'react-hook-form'

export interface Props<T extends FieldValues = FieldValues> {
	control: Control<T> | undefined
	readonly name: Path<T>
	readonly label: string
	readonly placeHolder?: string
	readonly className?: string
}
