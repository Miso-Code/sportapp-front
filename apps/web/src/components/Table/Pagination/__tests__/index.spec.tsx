import TablePaginationActions from '..'
import {
	act,
	fireEvent,
	render,
	RenderResult,
	waitFor
} from '@testing-library/react'
import { TablePaginationActionsProps } from '../interfaces'

describe('TablePaginationActions', () => {
	let wrapper: RenderResult
	let props: TablePaginationActionsProps

	beforeEach(() => {
		props = {
			count: 10,
			page: 1,
			rowsPerPage: 5,
			onPageChange: jest.fn()
		}
		wrapper = render(<TablePaginationActions {...props} />)
	})

	it('should render correctly', () => {
		expect(wrapper.container).toMatchSnapshot()
	})

	it('should call onPageChange with 0 when handleFirstPageButtonClick is called', () => {
		wrapper.getByLabelText('first page').click()
		expect(props.onPageChange).toHaveBeenCalledWith(expect.anything(), 0)
	})

	it('should call onPageChange with page - 1 when handleBackButtonClick is called', () => {
		wrapper.getByLabelText('previous page').click()
		expect(props.onPageChange).toHaveBeenCalledWith(
			expect.anything(),
			props.page - 1
		)
	})

	it('should call onPageChange with page + 1 when handleNextButtonClick is called', () => {
		const button = wrapper.container.querySelector(
			'button[aria-label="next page"]'
		)
		act(() => {
			fireEvent.click(button as Element)
		})

		waitFor(() => {
			expect(props.onPageChange).toHaveBeenCalledWith(
				expect.anything(),
				props.page + 1
			)
		})
	})

	it('should call onPageChange with Math.max(0, Math.ceil(count / rowsPerPage) - 1) when handleLastPageButtonClick is called', () => {
		const button = wrapper.container.querySelector(
			'button[aria-label="last page"]'
		)
		act(() => {
			fireEvent.click(button as Element)
		})

		waitFor(() => {
			expect(props.onPageChange).toHaveBeenCalledWith(
				expect.anything(),
				Math.max(0, Math.ceil(props.count / props.rowsPerPage) - 1)
			)
		})
	})

	it('should call onPageChange with 0 when handleFirstPageButtonClick is called', () => {
		wrapper.getByLabelText('first page').click()
		expect(props.onPageChange).toHaveBeenCalledWith(expect.anything(), 0)
	})

	it('should call onPageChange with page - 1 when handleBackButtonClick is called', () => {
		wrapper.getByLabelText('previous page').click()
		expect(props.onPageChange).toHaveBeenCalledWith(
			expect.anything(),
			props.page - 1
		)
	})

	it('calls onPageChange with the next page when handleNextButtonClick is called', () => {
		const onPageChange = jest.fn()
		const propsAux = {
			...props,
			onPageChange
		}
		wrapper.rerender(<TablePaginationActions {...propsAux} />)
		const nextButton = document.querySelector(
			'button[aria-label="next page"]'
		)
		act(() => fireEvent.click(nextButton!))

		waitFor(() =>
			expect(onPageChange).toHaveBeenCalledWith(
				expect.anything(),
				propsAux.page + 1
			)
		)
	})
})
