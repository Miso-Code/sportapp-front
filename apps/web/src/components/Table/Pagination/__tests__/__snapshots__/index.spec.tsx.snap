// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TablePaginationActions should render correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-1zye22"
  >
    <button
      aria-label="first page"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="FirstPageIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18.41 16.59 13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="previous page"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="KeyboardArrowLeftIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      aria-label="next page"
      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="KeyboardArrowRightIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6z"
        />
      </svg>
    </button>
    <button
      aria-label="last page"
      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LastPageIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5.59 7.41 10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
        />
      </svg>
    </button>
  </div>
</div>
`;
